web/src/components/data-entities/diver/DiverList.js

```diff
@@ -1,103 +1,129 @@
-import React, {useEffect, useState} from 'react';
+import React, {useState, useRef} from 'react';
 import {Box, Button, Typography} from '@mui/material';
 import {NavLink} from 'react-router-dom';
 import {grey, red} from '@mui/material/colors';
 import {getResult, entityEdit} from '../../../api/api';
 import LoadingOverlay from '../../overlays/LoadingOverlay';
 import {AgGridColumn, AgGridReact} from 'ag-grid-react';
 import {Add, Save} from '@mui/icons-material';
+import {constants} from '../../../common/constants';
 import 'ag-grid-enterprise';
 
 const DiverList = () => {
-  const [gridApi, setGridApi] = useState();
-  const [delta, setDelta] = useState([]);
-  const [rowData, setRowData] = useState();
-  const [errors, setErrors] = useState([]);
+    const [delta, setDelta] = useState([]);
+    const [errors, setErrors] = useState([]);
+    const dGridRef = useRef(null);
 
-  useEffect(() => {
-    async function fetchDivers() {
-      await getResult('divers').then((res) => setRowData(res.data));
-    }
-    fetchDivers();
-  }, []);
+    // Auto size function to be call each time data changed, so the grid always autofit
+    const autoSizeAll = (skipHeader) => {
+        if (dGridRef.current != null) {
+            dGridRef.current.columnApi.autoSizeAllColumns(skipHeader);
+        }
+    };
 
-  useEffect(() => {
-    if (gridApi) gridApi.redrawRows();
-  }, [gridApi, delta, errors]);
+    const onGridReady = (event) => {
+        async function fetchDivers(event) {
+            await getResult(constants.diverList.URL).then(
+                (res) => {
+                    // Use setRowData in api will not trigger onGridReady but onDataChange event.
+                    // if you use useState and connect row to setRowData then you will
+                    // keep fire onGridReady as row change
+                    console.debug('Loading diver dataset..');
+                    console.debug(res);
+                    event.api.setRowData(res.data);
+                    console.debug('Done loading diver dataset..');
+                });
+        }
 
-  const onCellValueChanged = (e) => {
-    setDelta((data) => {
-      const newData = {...data};
-      newData[e.data.diverId] = e.data;
-      return newData;
-    });
-  };
+        fetchDivers(event).then(() => {
+            // Now we have the data to do auto sizing, however the build in function only auto size visible rows,
+            // so when user scroll we need to auto size again
+            console.debug('Auto sizing after dataset loaded');
+            autoSizeAll(false);
+        });
+    };
 
-  const chooseCellStyle = (params) => {
-    if (params.context.errors.some((e) => e.id === params.data.diverId)) return {backgroundColor: red[100]};
-    if (params.context.delta[params.data.diverId]) return {backgroundColor: grey[100]};
-    return null;
-  };
+    // Why need this extra redraw?
```

Comment:
Just delete code that isn't necessary. Avoid leaving code commented out if possible.

Reply:
Done